api/src/test/java/org/openmrs/module/ModuleUtilTest.java

```diff
@@ -651,4 +653,120 @@ public void getResourceFromApi_shouldReturnNullIfFileIsNotFoundInApi() throws Ex
+		File destinationFolder = this.getEmptyJarDestinationFolder();
+
+		ModuleUtil.expandJar(getJarFile(), destinationFolder, directoryPath, false);
+
+		List<File> actualExpandedFiles = (List<File>)FileUtils.listFiles(destinationFolder, null, true);
+		Assert.assertEquals(numberOfFilesInSpecifiedDirectory, actualExpandedFiles.size());
+		Assert.assertEquals(destinationFolder.toString(), actualExpandedFiles.get(0).getParent());
+
+		FileUtils.deleteDirectory(destinationFolder);
+	}
+
+	/**
+	 * @see ModuleUtil#expandJar(File,File,String,boolean)
+	 * @verifies expand entire jar if name is empty string
+	 */
+	@Test
+	public void expandJar_shouldExpandEntireJarIfNameIsEmptyString() throws Exception {
+		final int numberOfFilesInJar = 6;
+		File destinationFolder = this.getEmptyJarDestinationFolder();
+
+		ModuleUtil.expandJar(getJarFile(), destinationFolder, "", false);
+
+		Assert.assertEquals(numberOfFilesInJar, FileUtils.listFiles(destinationFolder, null, true).size());
+
+		FileUtils.deleteDirectory(destinationFolder);
+	}
+
+	/**
+	 * @see ModuleUtil#expandJar(File,File,String,boolean)
+	 * @verifies expand entire jar if name is null
+	 */
+	@Test
+	public void expandJar_shouldExpandEntireJarIfNameIsNull() throws Exception {
+		final int numberOfFilesInJar = 6;
+		File destinationFolder = this.getEmptyJarDestinationFolder();
+
+		ModuleUtil.expandJar(getJarFile(), destinationFolder, null, false);
+
+		Assert.assertEquals(numberOfFilesInJar, FileUtils.listFiles(destinationFolder, null, true).size());
+
+		FileUtils.deleteDirectory(destinationFolder);
+	}
+
+	/**
+	 * @see ModuleUtil#expandJar(File,File,String,boolean)
+	 * @verifies expand file with parent tree if name is file and keepFullPath is true
+	 */
+	@Test
+	public void expandJar_shouldExpandFileWithParentTreeIfNameIsFileAndKeepFullPathIsTrue() throws Exception {
+		String fileName = "META-INF/maven/org.openmrs.module/test1-api/pom.properties";
+		File destinationFolder = this.getEmptyJarDestinationFolder();
+
+		ModuleUtil.expandJar(getJarFile(), destinationFolder, fileName, true);
+
+		List<File> actualExpandedFiles = (List<File>)FileUtils.listFiles(destinationFolder, null, true);
+		Assert.assertEquals(1, actualExpandedFiles.size());
+		File expectedPath = new File(destinationFolder, fileName);
+		Assert.assertEquals(expectedPath.toString(), actualExpandedFiles.get(0).toString());
+
+		FileUtils.deleteDirectory(destinationFolder);
+	}
+
+	/**
+	 * Gets Jar file to be expanded.
+	 * 
+	 * @return <code>File</code> containing Jar file.
+	 */
+	protected File getJarFile() {
+		return new File(this.getClass().getResource("/org/openmrs/module/include/testJarExpand.omod").getFile());
+	}
+	
+	/**
+	 * Gets folder to which Jar should be extracted. 
+	 * 
+	 * @return <code>File</code> containing folder for Jar tests.
+	 */
+	protected File getEmptyJarDestinationFolder() throws IOException {
+		File destinationFolder = new File("/tmp/expandedJar");
```

Comment:
I would need another windows user to run and confirm it. :)
Why don't you use System.getProperty("java.io.tmpdir")?